llvm/lib/Transforms/Utils/PromoteMemoryToRegister.cpp

```diff
@@ -1113,6 +1113,14 @@ void PromoteMem2Reg::RenamePass(BasicBlock *BB, BasicBlock *Pred,
         for (unsigned i = 0; i != NumEdges; ++i)
           APN->addIncoming(IncomingVals[AllocaNo], Pred);
 
+        if (APN->isComplete() &&
+            APN->getFunction()
+                ->getFnAttribute("no-signed-zeros-fp-math")
+                .getValueAsBool() &&
+            isa<FPMathOperator>(APN)) {
+          APN->setHasNoSignedZeros(true);
```

Comment:
I'd probably reorder these checks; FPOp = dyn_cast<FPMathOperator>(); if (FPop && fnAttrIsNSZ) { FPOp->setHasNoSignedZeros() } 

This also needs a comment explaining why this needs to be done

Reply:
`FPOp->setHasNoSignedZeros()` was not allowed since setHasNoSignedZeros() is a private member of FPMathOperator

Reply:
It doesn't really make sense that it would be public on Instruction but not on FPMathOperator